crates/net/ecies/src/stream.rs

```diff
@@ -40,6 +46,27 @@ where
         transport: Io,
         secret_key: SecretKey,
         remote_id: PeerId,
+    ) -> Result<Self, ECIESError> {
+        Self::connect_with_timeout(transport, secret_key, remote_id, HANDSHAKE_TIMEOUT).await
+    }
+
+    /// Wrapper around connect_no_timeout which enforces a timeout.
+    pub async fn connect_with_timeout(
+        transport: Io,
+        secret_key: SecretKey,
+        remote_id: PeerId,
+        timeout_limit: Duration,
+    ) -> Result<Self, ECIESError> {
+        timeout(timeout_limit, Self::connect_no_timeout(transport, secret_key, remote_id))
+            .await
+            .map_err(|_| ECIESError::from(ECIESErrorImpl::StreamTimeout))?
+    }
+
+    /// Connect to an `ECIES` server with no timeout.
+    pub async fn connect_no_timeout(
```

Comment:
`without` seems slightly better to me, can't think of anything more concise though:

```suggestion
    /// Connect to an `ECIES` server with no timeout.
    pub async fn connect_without_timeout(
```

Reply:
agree, have changed for handshake as well.